src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java

```diff
@@ -58,4 +61,81 @@ public void testGetDatabaseFromMatchesDatabaseWithEntriesWithCorrectMatch() {
         List<BibEntry> matches = new DatabaseSearcher(new SearchQuery("harrer", true, true), database).getMatches();
         assertEquals(Collections.singletonList(entry), matches);
     }
+
+    @Test
+    public void testGetDatabasefromMatchesEmptyDatabase() {
+        BibDatabase database = new BibDatabase();
+        SearchQuery query = new SearchQuery("asdf", true, true);
+
+        DatabaseSearcher databaseSearcher = new DatabaseSearcher(query, database);
+
+        assertTrue(databaseSearcher.getMatches().isEmpty());
```

Comment:
getMatches probably returns a list and thus it is better to compare against `Collections.emptyList()`